### PR TITLE
UofT-DSI | deep_learning - assignment 1

### PR DESCRIPTION
**Pull request by DmytroBolokhonov**
## What changes are you trying to make? (e.g. Adding or removing code, refactoring existing code, adding reports)
Adding data augmentation and transfer learning techniques to improve the performance of a CNN model for image classification.

## What did you learn from the changes you have made?
I considered fine-tuning additional layers of the pre-trained model and experimenting with different optimizers or loss functions to handle class imbalances.

## Was there another approach you were thinking about making? If so, what approach(es) were you thinking of?
I considered fine-tuning additional layers of the pre-trained model and experimenting with different optimizers or loss functions to handle class imbalances.

## Were there any challenges? If so, what issue(s) did you face? How did you overcome it?
The main challenge was ensuring proper integration of transfer learning and handling misclassified images. These were addressed by freezing early layers of the pre-trained model and using data augmentation to improve generalization.

## How were these changes tested?
The changes were tested by training the model on the augmented dataset, validating on unseen data, and evaluating test accuracy, precision, recall, and F1-score.

## Checklist
- [x] I can confirm that my changes are working as intended

### CODE REVIEW

**Review by tianyi21 — APPROVED** [review]
Hi Dmytro!

Thank you for submitting your DL assignments. Your assignment 1 is now complete.

The following comment(s) is for your reference, and there is no need to revise and resubmit your assignment:
* I saw you provided a thorough and complete discussion in this assignment. I believe you have a clear understanding of the problem you're working on and the design choices you have made, but please kindly remember to answer those short open-ended questions hidden between questions (e.g. Task 3a, Task 3b) in future assignments.

Thanks,
Tianyi [LS]
